packages/amplify-cli/src/extensions/amplify-helpers/get-env-info.ts

```diff
@@ -3,12 +3,23 @@ import { $TSAny, AmplifyError, stateManager } from 'amplify-cli-core';
 /**
  * returns the current environment info
  */
+let exceptionPrinted = false;
+
 export const getEnvInfo = (): $TSAny => {
+try {
   if (stateManager.localEnvInfoExists()) {
     return stateManager.getLocalEnvInfo();
   }
   throw new AmplifyError('EnvironmentNotInitializedError', {
     message: 'Current environment cannot be determined.',
     resolution: `Use 'amplify init' in the root of your app directory to create a new environment.`,
-  });
+    cause: new Error('`amplify init` not done'),
+  }); 
+  // handle error message gracefully by swallowing the stack trace
+} catch(error) {
+  if (!exceptionPrinted) {
+    exceptionPrinted = true;
+    console.error(`Error: ${error.message}\nResolution: ${error.resolution}`);
+  } 
+}
```

Comment:
We should not make this change here. `getEnvInfo` is called in multiple scenarios - we should not swallow the exception there.
If you're attempting to fix amplify mock api then I suggest you catch that error in the mock package and apply special handling there.

Reply:
yes, I am attempting to make changes to the mock. Sure, thanks @sobolk, will make changes there : )